src/main/scala/org/apache/spark/mllib/optimization/tfocs/LinearOperator.scala

```diff
@@ -0,0 +1,111 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.mllib.optimization.tfocs
+
+import org.apache.spark.mllib.linalg.BLAS
+import org.apache.spark.mllib.linalg.{ DenseVector, Vector, Vectors }
+import org.apache.spark.mllib.optimization.tfocs.CheckedIteratorFunctions._
+import org.apache.spark.mllib.optimization.tfocs.VectorSpace._
+
+/**
+ * A trait for linear operators supporting application of an operator and of its adjoint.
+ *
+ * @tparam X Type representing an input vector.
+ * @tparam Y Type representing an output vector.
+ */
+trait LinearOperator[X, Y] {
+  /**
+   * Evaluates this operator at x.
+   */
+  def apply(x: X): Y
+
+  /**
+   * The adjoint of this operator.
+   */
+  def a: LinearOperator[Y, X]
```

Comment:
`t` is more common to denote `transpose` or `adjoint`. `a` is commonly used in other places.